### PR TITLE
Add Frame creation tests for iterable columns or index

### PR DESCRIPTION
**Pull request by colinalexander**

### CODE REVIEW

File: static_frame/test/unit/test_frame.py [hunk 1]
```diff
@@ -143,6 +143,20 @@ def test_frame_init_i(self):
         self.assertEqual(f1.to_pairs(0),
             (('x', (('a', 3), ('b', 4), ('c', 5))), ('y', (('a', 12), ('b', 10), ('c', 11)))))
 
+    def test_frame_init_iter(self):
+
+        f1 = Frame(None, index=range(3), columns=("A",))
```
Comment (brandtbucher) [inline static_frame/test/unit/test_frame.py]:
```suggestion
        f1 = Frame(None, index=iter(range(3)), columns=("A",))
```

File: static_frame/test/unit/test_frame.py [hunk 2]
```diff
@@ -143,6 +143,20 @@ def test_frame_init_i(self):
...
+        self.assertEqual(
+            f1.to_pairs(0),
+            (('A', ((0, None), (1, None), (2, None))),)
+        )
+
+        f2 = Frame(None, index=("A",), columns=range(3))
```
Comment (brandtbucher) [inline static_frame/test/unit/test_frame.py]:
```suggestion
        f2 = Frame(None, index=("A",), columns=iter(range(3)))
```

File: static_frame/test/unit/test_frame.py (hunk 1 above)
Comment (colinalexander) [reply static_frame/test/unit/test_frame.py]: done.

File: static_frame/test/unit/test_frame.py (hunk 2 above)
Comment (colinalexander) [reply static_frame/test/unit/test_frame.py]: done.

**Comment by flexatone** [comment]
Thanks for these additions!

**Pull request merged**